discovery-provider/src/tasks/tracks.py

```diff
@@ -205,6 +205,13 @@ def wrapper(*args, **kargs):
     return wrapper
 
 
+pending_track_routes: List[TrackRoute] = []
+
+
+def reset_pending_updates():
```

Comment:
could we "export" this method as something a little bit more clear (reset_pending_track_routes) and then call that from a function defined in index.py called "reset_pending_updates" that acts as kind of the "teardown fixture" for an indexing iteration?

wdyt?

Reply:
Did one better and made the state local to the `track_state_update` and plumbed it through per @piazzatron 's request